File: Assignment GTR Ltd/Frontend Angular/app/controllers/productmaster.js
var app = angular.module('my-app', []);
app.controller('masterCtrl',($scope,$http,$window)=>{
  $scope.productmasteradd = ()=>{
    var data = 
    {
        Name:$scope.Name, 
        Barcode:$scope.Barcode,
        Type:$scope.Type, 
        Quantity:$scope.Quantity,
        Price:$scope.Price, 
    };
    $http.post("https://localhost:7180/api/productmaster/add",data).then(function(res){
            alert("Product created successfully");
            $window.location.href = 'productmasterList.html';
        }).catch(function(err){
            alert("Fields are required");
            
        });
  };
});

File: Assignment GTR Ltd/Asp.net Core Web API Assignment Backend/Asp.net Core Web API Assignment Backend/Models/ProductMaster.cs
namespace Asp.net_Core_Web_API_Assignment_Backend.Models
{
    public class ProductMaster
    {
        //public ProductMaster()
        //{
            //this.ProductDetails = new HashSet<ProductDetail>();
        //}
        public int Id { get; set; }
        public string Name { get; set; }
        public string Barcode { get; set; }
        public string Type { get; set; }
        public int Quantity { get; set; }
        public double Price { get; set; }
        //public virtual List<ProductDetail> ProductDetails { get; set; }
    }
}
